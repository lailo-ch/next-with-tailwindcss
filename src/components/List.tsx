import * as React from 'react'
import clsx from 'clsx'

export interface Props {
  className?: string | Record<string, unknown>
  ordered?: boolean
  item?: boolean
}

const List: React.FC<Props> = ({
  className,
  item,
  children,
  ordered = false,
}) => {
  const sharedStyles = 'text-base text-gray-700 dark:text-gray-300'

  if (item) {
    return <li className={clsx('', sharedStyles, className)}>{children}</li>
  }

  const Component = ordered ? 'ol' : 'ul'
  return (
    <Component
      className={clsx(
        'list-inside leading-loose',
        { 'list-decimal': ordered, 'list-disc': !ordered },
        sharedStyles,
        className
      )}
    >
      {children}
    </Component>
  )
}

export default List
