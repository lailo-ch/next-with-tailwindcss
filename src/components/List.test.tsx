import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import List from './List'

describe('List', () => {
  it('matches empty ul snapshot', () => {
    const wrapper = shallow(<List>items</List>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches empty ol snapshot', () => {
    const wrapper = shallow(<List ordered>items</List>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches li snapshot', () => {
    const wrapper = shallow(<List item>item text</List>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches ul with item snapshot', () => {
    const wrapper = shallow(
      <List>
        <List item>item text</List>
        <List item>item text</List>
      </List>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches empty ol snapshot', () => {
    const wrapper = shallow(
      <List ordered>
        <List item>item text</List>
        <List item>item text</List>
      </List>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
