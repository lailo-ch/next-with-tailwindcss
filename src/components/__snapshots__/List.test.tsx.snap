// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List matches empty ol snapshot 1`] = `
<ol
  className="list-inside leading-loose list-decimal text-base text-gray-700 dark:text-gray-300"
>
  items
</ol>
`;

exports[`List matches empty ol snapshot 2`] = `
<ol
  className="list-inside leading-loose list-decimal text-base text-gray-700 dark:text-gray-300"
>
  <List
    item={true}
  >
    item text
  </List>
  <List
    item={true}
  >
    item text
  </List>
</ol>
`;

exports[`List matches empty ul snapshot 1`] = `
<ul
  className="list-inside leading-loose list-disc text-base text-gray-700 dark:text-gray-300"
>
  items
</ul>
`;

exports[`List matches li snapshot 1`] = `
<li
  className="text-base text-gray-700 dark:text-gray-300"
>
  item text
</li>
`;

exports[`List matches ul with item snapshot 1`] = `
<ul
  className="list-inside leading-loose list-disc text-base text-gray-700 dark:text-gray-300"
>
  <List
    item={true}
  >
    item text
  </List>
  <List
    item={true}
  >
    item text
  </List>
</ul>
`;
